File: 121_best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ret = 0;
        int min = 999999;
        int size = prices.size();
        for(int i = 0; i< size; ++i){
            if( min > prices[i] ) min = prices[i];
            else if ( prices[i] - min > ret ){
                ret = prices[i] - min;
            }
        }
        return ret;
    }
};

File: 121_best_time_to_buy_and_sell_stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        ret, min= 0, 999999
        for i in prices:
            if min>i:
                min = i
            elif i-min>ret:
                ret = i-min
        return ret
